Argus-Labs/world-engine
# fix(nakama): automatic websocket reconnect

Closes: GDEV-1024

## Overview

Automatically attempt to reconnect with a websocket when cardinal goes down.

## Testing and Verifying

Unit test added.

To manually test:

1. Run `docker compose up nakama game --build` from the world-engine root directory.
2. Once all the docker containers are running, run an en end-to-end test that makes use of receipts, e.g. `TestReceipts` in `world-engine/e2e/tests/nakama/nakama_test.go`. Notice that the test passes successfully.
3. Stop the `test_game` container
4. Notice that the `test_nakama` logs will complain with a "No host found: websocket dial failed" message.
5. Restart the `test_game` container.
7. Rerun the `TestReceipts` test and notice that it still passes successfully.
